### DISCUSSION TITLE
Unable to switch the tabs using mouse. Only switching the tab can be done through TabKey navigation at DesignTime

### DISCUSSION THREAD

**Discussion by Sekar47** [dotnet/winforms#12390] [General] [answered]
### Environment

Edition - Windows 11 Enterprise
Version - 23H2
OS build - 22631.3737
Visual Studio version - 17.9.2

### .NET version

.Netcore 6,7,8

### Did this work in a previous version of Visual Studio and/or previous .NET release?

No, it doesn't work on previous NetCore versions.

### Issue description

I'm trying to switch the tabs by using the mouse click at DesignTime but can't be able to switch. But can be able to be switching by using the TabKey.
For your reference, I have attached a simple sample that replicates the issue.
[CustomTabControl_Sample.zip](https://github.com/user-attachments/files/16142154/CustomTabControl_Sample.zip)

### Steps to reproduce

1.Switch the tabs by using the TabKey.
2.Switch the tabs by using the mouse.

Expected Behavior: While switch using the mouse, the selected tab needs to be switched.
Observed Behavior: No tab switches happened.

**Comment by elachlan** [comment]
@Olina-Zhang can your team please confirm?

**Comment by Zheng-Li01** [comment]
@elachlan  there have a known issue in the designer repo to track https://github.com/microsoft/winforms-designer/issues/3201.

**Comment by Sekar47** [comment]
Hi @elachlan ,
I'm facing 404 issue while trying to open the GitHub link you provided.

**Comment by elachlan** [comment]
Hi @Sekar47, the link provided is for the designer repo and it's currently private.

**Comment by merriemcgaw** [comment]
@Sekar47 we're investigating in our internal designer repo. We'll update this when we have an investigation.

**Comment by Sekar47** [comment]
Hi @merriemcgaw , Is there any update for this issue?

**Comment by merriemcgaw** [comment]
@Sekar47  not yet, I'm sorry.

**Comment by Sekar47** [comment]
Hi @merriemcgaw , Is there any update for this issue?

**Comment by merriemcgaw** [comment]
@Sekar47 this is on our backlog, but I don't have an estimate for when we'll have a fix. 

That said, we have confirmed that you can access the desired tab directly if you select it via the PropertyBrowser or DocumentOutline windows in Visual Studio. Given that there are multiple ways to get to any TabPage that you need, it may take us just a little longer to address this specific issue.

**Comment by Tanya-Solyanik** [comment]
@Sekar47  - mouse click is processed by the VisualStudio process and is forwarded to the DesignTOolsProcess as an input message to be processed by the corresponding ControlBodyGlyphs. 
https://learn.microsoft.com/en-us/dotnet/desktop/winforms/controls-design/designer-differences-framework?view=netdesktop-8.0
Also see samples here - [microsoft/winforms-designer-extensibility](https://github.com/microsoft/winforms-designer-extensibility)

**Comment by LeafShi1** [comment]
@Sekar47 I can't open the Form designer of your provided project, but in .net8.0/.net9.0 projects, the tabs can be switched by using mouse
![TabControl](https://github.com/user-attachments/assets/caa6e395-de05-4621-9fa8-34ff7b0e0ba6)
If I misunderstood you, please correct me

But this issue https://github.com/microsoft/winforms-designer/issues/3201 does exist. When the mouse is constantly clicked on the spinner, the tab cannot be switched continuously.

**Reply by Sekar47** [reply]
Hi @LeafShi1 ,

Thank you for your response.

I tested my sample, and while I can open the Form Designer, I still cannot switch tabs using mouse clicks. Switching works with the Tab key, but not with the mouse.

I’ve attached a video and the sample project for reference. Please let me know if further details are needed.

[CustomTabControl_Sample.zip](https://github.com/user-attachments/files/18004495/CustomTabControl_Sample.zip)
[TabSwitch-Video.zip](https://github.com/user-attachments/files/18004504/TabSwitch-Video.zip)

**Comment by Epica3055** [comment, marked as answer]
Try the following change
```c#
CustomTabControlDesigner.cs
        protected override bool GetHitTest(Point point)
        {
            if (tabControl != null)
            {
                for (int i = 0; i < tabControl.TabPages.Count; i++)
                {
                    Point clientPoint = tabControl.PointToClient(point);
                    if (tabControl.GetTabRect(i).Contains(clientPoint))
                    {
                        return true;
                    }
                }
            }
            return base.GetHitTest(point);
        }
```

**Reply by LeafShi1** [reply]
@Sekar47 This problem is because the mouse is positioned at the wrong point. Please add 
`Point clientPoint = tabControl.PointToClient(point);` in the `GetHitTest` method according to the code provided by Epica to verify this problem.
